### ISSUE TITLE
Canceling test connection does not really cancel

### ISSUE THREAD

**Issue by MikhailArkhipov** [Microsoft/RTVS#2747]
Start testing remote connection
Double-click on local

RTVS connects to local
Few seconds later certificate verification popup shows up

System.Net knows nothing about cancellation.

```
>	Microsoft.R.Host.Client.dll!Microsoft.R.Host.Client.Host.RemoteBrokerClient.ValidateCertificateHttpHandler(object sender, System.Security.Cryptography.X509Certificates.X509Certificate certificate, System.Security.Cryptography.X509Certificates.X509Chain chain, System.Net.Security.SslPolicyErrors sslPolicyErrors) Line 71	C#
 	System.dll!System.Net.ServerCertValidationCallback.Callback(object state)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state)	Unknown
 	System.dll!System.Net.ServerCertValidationCallback.Invoke(object request, System.Security.Cryptography.X509Certificates.X509Certificate certificate, System.Security.Cryptography.X509Certificates.X509Chain chain, System.Net.Security.SslPolicyErrors sslPolicyErrors)	Unknown
 	System.dll!System.Net.ServicePoint.HandshakeDoneProcedure.CertValidationCallback(string hostName, System.Security.Cryptography.X509Certificates.X509Certificate certificate, System.Security.Cryptography.X509Certificates.X509Chain chain, System.Net.Security.SslPolicyErrors sslPolicyErrors)	Unknown
 	System.dll!System.Net.Security.SecureChannel.VerifyRemoteCertificate(System.Net.Security.RemoteCertValidationCallback remoteCertValidationCallback)	Unknown
 	System.dll!System.Net.Security.SslState.CompleteHandshake()	Unknown
 	System.dll!System.Net.Security.SslState.CheckCompletionBeforeNextReceive(System.Net.Security.ProtocolToken message, System.Net.AsyncProtocolRequest asyncRequest)	Unknown
 	System.dll!System.Net.Security.SslState.StartSendBlob(byte[] incoming, int count, System.Net.AsyncProtocolRequest asyncRequest)	Unknown
 	System.dll!System.Net.Security.SslState.ProcessReceivedBlob(byte[] buffer, int count, System.Net.AsyncProtocolRequest asyncRequest)	Unknown
 	System.dll!System.Net.Security.SslState.ReadFrameCallback(System.Net.AsyncProtocolRequest asyncRequest)	Unknown
 	System.dll!System.Net.AsyncProtocolRequest.CompleteRequest(int result)	Unknown
 	System.dll!System.Net.FixedSizeReader.CheckCompletionBeforeNextRead(int bytes)	Unknown
 	System.dll!System.Net.FixedSizeReader.ReadCallback(System.IAsyncResult transportResult)	Unknown
 	System.dll!System.Net.LazyAsyncResult.Complete(System.IntPtr userToken)	Unknown
 	System.dll!System.Net.ContextAwareResult.CompleteCallback(object state)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state)	Unknown
 	System.dll!System.Net.ContextAwareResult.Complete(System.IntPtr userToken)	Unknown
 	System.dll!System.Net.LazyAsyncResult.ProtectedInvokeCallback(object result, System.IntPtr userToken)	Unknown
 	System.dll!System.Net.Sockets.BaseOverlappedAsyncResult.CompletionPortCallback(uint errorCode, uint numBytes, System.Threading.NativeOverlapped* nativeOverlapped)	Unknown
 	mscorlib.dll!System.Threading._IOCompletionCallback.PerformIOCompletionCallback(uint errorCode, uint numBytes, System.Threading.NativeOverlapped* pOVERLAP)	Unknown

```

**Comment by MikhailArkhipov** [comment]
ConnectionManagerViewModel.Connect was not canceling test connection. Added cancellation and also passed token down to RemoteBrokerClient so it can short-circuit certificate validation if cancellation was requested.

**Comment by MikhailArkhipov** [comment]
Most probably fixes #2742 as well

**Comment by MikhailArkhipov** [comment]
https://github.com/Microsoft/RTVS/commit/0c7df6d1552c3a36e431fad30fc338dd80acc2b9
